rubocop.yml

```diff
@@ -242,3 +248,45 @@ Style/HashLikeCase:
 
 Style/RedundantFileExtensionInRequire:
   Enabled: true
+
+Lint/BinaryOperatorWithIdenticalOperands:
+  Enabled: true
+
+Lint/DuplicateRescueException:
+  Enabled: true
+
+Lint/EmptyConditionalBody:
+  Enabled: true
+
+Lint/FloatComparison:
+  Enabled: true
+
+Lint/MissingSuper:
+  Enabled: true
+
+Lint/OutOfRangeRegexpRef:
+  Enabled: true
+
+Lint/SelfAssignment:
+  Enabled: true
+
+Lint/TopLevelReturnWithArgument:
+  Enabled: true
+
+Lint/UnreachableLoop:
+  Enabled: true
+
+Style/ExplicitBlockArgument:
+  Enabled: true
+
+Style/GlobalStdStream:
+  Enabled: true
+
+Style/SingleArgumentDig:
+  Enabled: true
+
+Style/OptionalBooleanParameter:
+  Enabled: false
+
+Style/StringConcatenation:
+  Enabled: false
```

Comment:
This feels like it's got many legitimate use cases in paysvc and other projects, but happy to enable it otherwise

Reply:
Generally using the concat operators is fine if you can guarantee what is being passed is a string, however using string interpolation will automatically call `to_s` on objects passed to it and is considered safer. Also, concating is [slower than interpolation](https://stackoverflow.com/a/60094040). I agree though that there can be some cases where concating is handy but interpolation seems safer? Not fussed either way tbh!

```ruby
[1] (main)> class Foo
[1] (main)*   def to_s
[1] (main)*     puts "called"
[1] (main)*   end
[1] (main)* end
=> :to_s
[2] (main)> Foo.new + " 123"
NoMethodError: undefined method `+' for #<Foo:0x00007fb0f0c29e50>
from (pry):6:in `__pry__'
[3] (main)> "#{Foo.new} 123"
called
=> "#<Foo:0x00007fb0f1258b50> 123"
```

Reply:
The performance/safety issues is less relevant here unless using this is generally a bad idea in 100% of the cases. The main question for this stuff is whether we need to disable it a significant amount of time we need it (seems to be the case in our larger projects, and from what I can tell, for legitimate use cases too)